pkg/kubelet/prober/prober.go

```diff
@@ -277,3 +285,30 @@ func (eic execInContainer) SetStderr(out io.Writer) {
 func (eic execInContainer) Stop() {
 	//unimplemented
 }
+
+// selectProbeIP selects a pod IP for probes from the specified PodStatus
+// based on the specified preferredIPFamily
+func selectProbeIP(status v1.PodStatus, preferredIPFamily kubeletconfigv1beta1.IPFamily) (probeIP string) {
+	probeIP = status.PodIP
+	if len(status.PodIPs) == 0 || preferredIPFamily == kubeletconfigv1beta1.IPFamilyNone {
+		return probeIP
+	}
+	for _, podIP := range status.PodIPs {
+		if utilnet.IsIPv6String(podIP.IP) && preferredIPFamily == kubeletconfigv1beta1.IPFamilyIPv6 {
+			return podIP.IP
+		} else if !utilnet.IsIPv6String(podIP.IP) && preferredIPFamily == kubeletconfigv1beta1.IPFamilyIPv4 {
+			return podIP.IP
+		}
+	}
+	return probeIP
+}
+
+func parseToIPFamily(ipFamily string) kubeletconfigv1beta1.IPFamily {
```

Comment:
shouldn't we just validate that it is == the known constants on the way in (config validation) and not bother doing caseless stuff deeper in the stack?

Reply:
IMHO, caseless configuration is more user friendly, so i did this.